syft/execution/type_wrapper.py

```diff
@@ -0,0 +1,261 @@
+import syft as sy
+from syft.workers.abstract import AbstractWorker
+from warnings import warn
+from typing import Union
+from syft_proto.execution.v1.type_wrapper_pb2 import NestedTypeWrapper as NestedTypeWrapperPB
+from syft_proto.execution.v1.type_wrapper_pb2 import ClassType as ClassTypePB
+
+class NestedTypeWrapper:
+    """
+        Class for input type serialization and type checking for nested structures.
+    """
+    def __init__(self, nested_type=None):
+        if nested_type:
+            self.serialized_nested_type = NestedTypeWrapper.serialize_nested_type(nested_type)
+        else:
+            self.serialized_nested_type = None
+
+    @staticmethod
+    def serialize_nested_type(input_arg: any) -> Union[list, tuple, dict, type]:
+        """
+            Function to serialize input of a function/Plan, including nested types.
+
+            Note: supported nested structures: list, tuple, dict with string keys.
+
+            Params:
+                input_arg: *args of a function or Plan.
+
+            Returns:
+                Union[list, tuple, dict, type]: Nested structured with types instead of objects.
+        """
+        if isinstance(input_arg, (list, tuple)):
+            result = [NestedTypeWrapper.serialize_nested_type(elem) for elem in input_arg]
+            return tuple(result) if isinstance(input_arg, tuple) else result
+
+        if isinstance(input_arg, dict):
+            serialized_dict = {k: NestedTypeWrapper.serialize_nested_type(v) for k, v in input_arg.items()}
+            return serialized_dict
+
+        return type(input_arg)
+
+    @staticmethod
+    def raise_typecheck_warn(obj_type: str, obj_name: str, build: str, call: str, path: str) -> None:
```

Comment:
Could these methods take the object as a param and extract the type/name/id/string repr to form the error messages?